diff --git a/lisp/net/tramp-sh.el b/lisp/net/tramp-sh.el
index bce7c323dad..4638e94f842 100644
--- a/lisp/net/tramp-sh.el
+++ b/lisp/net/tramp-sh.el
@@ -4930,11 +4930,5 @@ tramp-ssh-controlmaster-options
            ;; Hashed tokens are introduced in OpenSSH 6.7.  On macOS
            ;; we cannot use an absolute file name, it is too long.
            ;; See Bug#19702.
-	   (if (eq system-type 'darwin)
-	       (if (tramp-ssh-option-exists-p vec "ControlPath=tramp.%C")
-		   "tramp.%%C" "tramp.%%r@%%h:%%p")
-	     (expand-file-name
-	      (if (tramp-ssh-option-exists-p vec "ControlPath=tramp.%C")
-		  "tramp.%%C" "tramp.%%r@%%h:%%p")
-	      (or small-temporary-file-directory
-		  tramp-compat-temporary-file-directory))))
+	   (if (tramp-ssh-option-exists-p vec "ControlPath=tramp.%C")
+		   "tramp.%%C" "tramp.%%r@%%h:%%p"))

          ;; ControlPersist option is introduced in OpenSSH 5.6.
 	 (when (and (not (eq tramp-use-connection-share 'suppress))
